tests/integration/helpers.py

```diff
@@ -187,3 +206,149 @@ async def run_command(model_name: str, app_name: str, unit_num: int, command: li
         raise e
     return res.stdout
+
+
+def present_facade(
+    interface: str,
+    app_data: Dict = None,
+    unit_data: Dict = None,
+    role: Literal["provide", "require"] = "provide",
+    model: str = None,
+    app: str = "facade",
+):
+    """Set up the facade charm to present this data over the interface ``interface``."""
+    data = {
+        "endpoint": f"{role}-{interface}",
+    }
+    if app_data:
+        data["app_data"] = json.dumps(app_data)
+    if unit_data:
+        data["unit_data"] = json.dumps(unit_data)
+
+    with tempfile.NamedTemporaryFile(dir=os.getcwd()) as f:
+        fpath = Path(f.name)
+        fpath.write_text(yaml.safe_dump(data))
+
+        _model = f" --model {model}" if model else ""
+
+        subprocess.run(shlex.split(f"juju run {app}/0{_model} update --params {fpath.absolute()}"))
+
+
+async def get_unit_address(ops_test: OpsTest, app_name, unit_no):
+    status = await ops_test.model.get_status()
+    app = status["applications"][app_name]
+    if app is None:
+        assert False, f"no app exists with name {app_name}"
+    unit = app["units"].get(f"{app_name}/{unit_no}")
+    if unit is None:
+        assert False, f"no unit exists in app {app_name} with index {unit_no}"
+    return unit["address"]
+
+
+async def deploy_and_configure_minio(ops_test: OpsTest):
+    config = {
+        "access-key": ACCESS_KEY,
+        "secret-key": SECRET_KEY,
+    }
+    await ops_test.model.deploy(MINIO, channel="edge", trust=True, config=config)
+    await ops_test.model.wait_for_idle(apps=[MINIO], status="active", timeout=2000)
+    minio_addr = await get_unit_address(ops_test, MINIO, "0")
+
+    mc_client = Minio(
+        f"{minio_addr}:9000",
+        access_key="accesskey",
+        secret_key="secretkey",
+        secure=False,
+    )
+
+    # create tempo bucket
+    found = mc_client.bucket_exists(BUCKET_NAME)
+    if not found:
+        mc_client.make_bucket(BUCKET_NAME)
+
+    # configure s3-integrator
+    s3_integrator_app: Application = ops_test.model.applications[S3_INTEGRATOR]
+    s3_integrator_leader: Unit = s3_integrator_app.units[0]
+
+    await s3_integrator_app.set_config(
+        {
+            "endpoint": f"minio-0.minio-endpoints.{ops_test.model.name}.svc.cluster.local:9000",
+            "bucket": BUCKET_NAME,
+        }
+    )
+
+    action = await s3_integrator_leader.run_action("sync-s3-credentials", **config)
+    action_result = await action.wait()
+    assert action_result.status == "completed"
+
+
+async def deploy_cluster(ops_test: OpsTest, tempo_app=APP_NAME):
+    # await ops_test.model.deploy(FACADE, channel="edge")
+    # await ops_test.model.wait_for_idle(
+    #     apps=[FACADE], raise_on_blocked=True, status="active", timeout=2000
+    # )
+
+    # TODO: deploy from latest edge
+    tempo_worker_charm = "/home/michael/Work/tempo-worker-k8s-operator/charm"
+
+    resources = {
+        "tempo-image": "docker.io/ubuntu/tempo:2-22.04",
```

Comment:
Could we use the image from the metadata?

Reply:
No. Since the metadata will be for the Tempo coordinator charm not the worker. Anyways, this line will be removed since we'll deploy worker from edge.